### PR TITLE
Ignore dynamically stored labels if Glean is not initialized 

### PR DESCRIPTION
**Pull request by badboy**

### CODE REVIEW

File: glean-core/src/metrics/labeled.rs [hunk 1]
```diff
@@ -177,10 +184,27 @@ where
...
-    pub fn get(&mut self, glean: &Glean, label: &str) -> T {
-        let label = match self.labels {
-            Some(_) => self.static_label(label),
-            None => self.dynamic_label(glean, label),
+    pub fn get<'a, G: Into<Option<&'a Glean>>>(&mut self, glean: G, label: &str) -> T {
+        // We have 3 scenarios to consider:
+        // * Static labels. No database access needed. We just look at what is in memory.
+        // * Dynamic labels, initialized Glean. We look up in the database all previously stored
+        //   labels in order to keep a maximum of allowed labels.
+        // * Dynamic labels, uninitialized Glean. We ignore potentially previously stored labels
```
Comment (Dexterp37) [inline glean-core/src/metrics/labeled.rs]: Can you reference [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1588451) here?

File: glean-core/android/src/test/java/mozilla/telemetry/glean/private/AccumulationsBeforeGleanInitTest.kt [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        )
+
+        labeledCounterMetric["label1"].add(1)
+
+        forceInitGlean()
+
+        assertEquals(1, labeledCounterMetric["label1"].testGetValue())
+    }
+
+    @Ignore
```
Comment (Dexterp37) [inline glean-core/android/src/test/java/mozilla/telemetry/glean/private/AccumulationsBeforeGleanInitTest.kt]: Can you add a note here? `@Ignore("Bug 1588452 will enable it back")` or something like that

File: glean-core/android/src/test/java/mozilla/telemetry/glean/private/AccumulationsBeforeGleanInitTest.kt [hunk 3]
```diff
@@ -33,6 +34,8 @@ class AccumulationsBeforeGleanInitTest {
     @Before
     fun cleanup() {
         Glean.testDestroyGleanHandle()
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.setTaskQueueing(true)
```
Comment (Dexterp37) [inline glean-core/android/src/test/java/mozilla/telemetry/glean/private/AccumulationsBeforeGleanInitTest.kt]: nit: can you add a comment (the one from the commit message is ok) before this line?

This should probably happen in `Glean.resetGlean` as well for other tests. Good catch. Would you mind filing a bug for it?

Comment (badboy) [reply glean-core/android/src/test/java/mozilla/telemetry/glean/private/AccumulationsBeforeGleanInitTest.kt]: https://bugzilla.mozilla.org/show_bug.cgi?id=1588463

Comment (mdboom) [inline glean-core/android/src/test/java/mozilla/telemetry/glean/private/AccumulationsBeforeGleanInitTest.kt]: We may also want to clear the queue to be extra sure...

Comment (badboy) [reply glean-core/android/src/test/java/mozilla/telemetry/glean/private/AccumulationsBeforeGleanInitTest.kt]: Yes, but blindly clearing breaks other tests. We should take that in a follow-up, see the filed bug for it.

Comment (mdboom) [reply glean-core/android/src/test/java/mozilla/telemetry/glean/private/AccumulationsBeforeGleanInitTest.kt]: "Fun"

**Pull request merged**